### DISCUSSION TITLE
GraphHopper (offline) routing future

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#637] [Q&A] [answered]
Does anyone use GraphHopper (offline) routing?

GraphHopper has stopped supporting offline routing and Android a long time ago:
- https://github.com/graphhopper/graphhopper/issues/1940

Cruiser uses an old GraphHopper version and it causes problems in maintenance.

So I'm thinking if it's really worth the effort or better to focus on **BRouter (offline)**
which has the advantages of cross-country routing and too many routing options.

**Comment by kds112** [comment]
I don't use it, you can concentrate just on the main services...

**Comment by GuG1050** [comment]
I do not use it.

**Comment by ramondevic** [comment]
I use It.

**Comment by ToToGSARider** [comment]
I download the offline maps to my cell phone so that I can then navigate. Is this function no longer available in Cruiser 5? 

I plan the routes in Kurviger and then use this map material for navigation. How should I understand this?

Ich lade mir die Offline Karten auf das Handy um dann navigieren zu können. Ist diese Funktion in Cruiser 5 jetzt auch nicht mehr verfügbar? 

Ich plane die Routen in kurviger und verwende dann zur Navigation dieses Kartenmaterial. Wie darf ich das jetzt verstehen?

![Screenshot_20241006_215800_Cruiser.jpg](https://github.com/user-attachments/assets/19483c7a-ac8e-4532-a807-73860f202fda)

**Reply by Ciutnik** [reply]
@ToToGSARider 
This thread is about withdrawing offline GraphHopper data from Cruiser. It is used ONLY FOR ROUTING - the same as the commonly used BRouter.

What you are showing is the map download menu, which is used ONLY TO DISPLAY THE MAP on the screen. You view and navigate against this map, but according to the route calculated by GraphHopper or BRouter, or several other routers (route finders) - offline or online.

So these are two different things, although closely related 🙂

**Reply by ToToGSARider** [reply]
@Ciutnik 
Okay, diese Antwort verwirrt mich jetzt total. Kann ich die Karten in Zukunft noch laden oder nicht? Werden diese Karten durch das neue Thema von MotoUKRider ersetzt? Also wenn mich jetzt jemand fragen würde: nutzt du die Daten? Würde ich sagen ja, da ich ja da auch mal Ziele im Umkreis suche. Ich ziehe mir mal die neuste Beta auf mein Testhandy und schaue was da so passiert. Im Cruiser 4.0.2bleibt ja alles beim alten denke ich. Ich werde sowieso frühestens umstellen, wenn im Cruiser 5 eine 3D Option vorhanden ist.

Okay, I'm totally confused by this answer. Can I still load the maps in the future or not? Will these maps be replaced by the new MotoUKRider theme? So if someone were to ask me now: do you use the data? I would say yes, as I sometimes search for destinations in the surrounding area. I'll pull the latest beta onto my test phone and see what happens. In Cruiser 4.0.2 everything stays the same I think. I won't change over until there is a 3D option in Cruiser 5 at the earliest anyway.

**Reply by Ciutnik** [reply]
Yes, of course you will always be able to load maps! But they are only a "background" for the route drawn by the router. And back to the topic: do you use GraphHopper (offline) to calculate the route? The default, suggested and recommended is BRouter (offline). If you haven't changed this in Cruiser settings - and I'm guessing you haven't - then nothing will change for you.

But you touched on another aspect - the theme of the map. For some time now the default has been MotoUKRider, which is now simply called "Light" or "Dark". It has an impact, to put it simply, ONLY on the map's color scheme (i.e. how it looks on the screen). And that is completely different from the topic of the thread (GraphHopper it's just a router).

At first glance, this may seem confusing, but it really isn't 🙂 @devemux86 has already done a lot to simplify all this: you can download maps, POIs, routing data, and a default theme with just one click.

Anticipating any potential doubts: altitude data is yet another matter... 🙂

_(sorry I only write in English... or rather "English" 😉)_

**Reply by ToToGSARider** [reply]
Okay, hier meine Vorgehensweise nach Rücksprache mir @devemux86 : Ich plane meine Routen ausschliesslich in kurviger und importiere dann die .kurviger Datei in die Cruiser App. Die Routenneuberechnung ist bei mir immer deaktiviert und ich habe es nach Rücksprache mit @devemux86  auf Valhalla umgestellt. Bin ich dann überhaupt vom Thema betroffen? Ich schreibe immer mit dem deepl Übersetzer und ziehe mir die Texte, wo ich denke, dass Infos für mich dabei sein könnten in den deepL Übersetzer. 

Okay, here's my approach after consulting @devemux86  : I plan my routes exclusively in kurviger and then import the .kurviger file into the Cruiser app. Route recalculation is always deactivated for me and I have changed it to Valhalla after consulting @devemux86 . Am I then affected by the issue at all? I always write with the deepl translator and drag the texts where I think there might be information for me into the deepL translator.

**Reply by Ciutnik** [reply]
It's simple: if you use Valhalla, you don't use GraphHopper.
So this issue does not concern you 🙂

**Reply by devemux86** [reply]
@ToToGSARider 

> Am I then affected by the issue at all?

No, you are not affected by this change.

Nein, Sie sind von dieser Änderung nicht betroffen.

> I have changed it to Valhalla

Valhalla needs internet. If you want offline rerouting, you can select **BRouter (offline)**.

Valhalla braucht Internet. Wenn Sie ein Offline-Rerouting wünschen, können Sie **BRouter (offline)** wählen.

**Reply by bios4** [reply]
To make it easily understandable - Settings > Routing > Advanced > Service will look like this after the change 😉:

<img src="https://github.com/user-attachments/assets/a7142d97-eb67-4f7e-b7a2-a48f6ba270ca" width="33%">

**Reply by ToToGSARider** [reply]
@devemux86 @Ciutnik 
Danke für die Aufklärung. Entschuldigung für die ganzen Fragen, aber der Cruiser bietet eben mittlerweile so viele Möglichkeiten, dass man eben schnell überfordert ist mit den ganzen Einstellungen. Es ist eben so, dass wenn man einmal sein Setup / Profil gefunden hat, dieses natürlich nicht unbedingt verändern möchte. Dann warte ich mal ab was die Zukunft bringt.

Thanks for the clarification. Sorry for all the questions, but the Cruiser now offers so many options that it's easy to get overwhelmed with all the settings. It's just that once you've found your setup / profile, you don't necessarily want to change it. I'll wait and see what the future brings.

**Reply by ramondevic** [reply]
I do use GraphHopper offline and I'm very happy with it.
I've tried BRouter offline but it doesn't show me the data I need: name of the destination, name of the street/road I'm on right now, and name of the next street/road.

PLEASE, DON'T GRAPHHOPPER DISAPPEAR!

**Reply by bios4** [reply]
Do not punish the messenger @ramondevic (because of your downvote of my post with the screenshot) 😉

**Comment by devemux86** [comment, marked as answer]
- Removed in https://github.com/devemux86/cruiser/discussions/641
